src/DistributionApi.php

```diff
@@ -62,14 +67,15 @@ public function __construct(
         $this->moduleDataProvider = $moduleDataProvider;
         $this->prestashopVersion = $prestashopVersion;
         $this->downloadDirectory = rtrim($downloadDirectory, '/');
+        $this->shopDataProvider = $shopDataProvider;
```

Comment:
I remember @lartist faced issues before with this change because of this scenario

1. Load Upgrade BO page using ps_distributionapiclient current version, this loads this PHP service from YAML definition (without `$shopDataProvider`)
2. Upgrade module version shopDataProvider to current to new: this PHP file changes
3. The PHP file has changed but the YAML definition has been cached so Symfony error happens (because it attempts to instantiate this module using old YAML definition without `$shopDataProvider`)

However https://github.com/PrestaShop/PrestaShop/pull/31137 is supposed to solve this problem -> you might want to just verify 😉

Reply:
Ok, I think we can ask to the QA to test this scenario :)

Reply:
It will be tested by QA when the module release is being tested. When QA team tests module upgrades from ps_distributionapiclient 1.0.3 to 1.1.0 they will see whether it crashes